pkg/resources/password_policy.go

```diff
@@ -0,0 +1,447 @@
+		IfNotExists:   d.Get("if_not_exists").(bool),
+		IfNotExistsOk: manager.Ok(d.GetOk("if_not_exists")),
+	}
+
+	stmt, err := manager.Create(input)
+	if err != nil {
+		return fmt.Errorf("couldn't generate create statement: %w", err)
+	}
+
+	db := meta.(*sql.DB)
+	_, err = db.Exec(stmt)
+	if err != nil {
+		return fmt.Errorf("error executing create statement: %w", err)
+	}
+
+	d.SetId(PasswordPolicyID(&input.PasswordPolicy))
+
+	return nil
+}
+
+// ReadPasswordPolicy implements schema.ReadFunc.
+func ReadPasswordPolicy(d *schema.ResourceData, meta interface{}) error {
+	manager, err := snowflake.NewPasswordPolicyManager()
+	if err != nil {
+		return fmt.Errorf("couldn't create password policy builder: %w", err)
+	}
+
+	input := &snowflake.PasswordPolicyReadInput{
+		Database:   d.Get("database").(string),
+		Schema:     d.Get("schema").(string),
+		ObjectName: d.Get("name").(string),
+	}
+
+	stmt, err := manager.Read(input)
+	if err != nil {
+		return fmt.Errorf("couldn't generate describe statement: %w", err)
+	}
+
+	db := meta.(*sql.DB)
+	rows, err := db.Query(stmt)
+	if err != nil {
+		return fmt.Errorf("error querying password policy: %w", err)
+	}
+
+	defer rows.Close()
+	output, err := manager.Parse(rows)
+	if err != nil {
+		return fmt.Errorf("failed to parse result of describe: %w", err)
+	}
+
+	if err = d.Set("min_length", output.MinLength); err != nil {
+		return fmt.Errorf("error setting min_length: %w", err)
+	}
+	if err = d.Set("max_length", output.MaxLength); err != nil {
+		return fmt.Errorf("error setting max_length: %w", err)
+	}
+	if err = d.Set("min_upper_case_chars", output.MinUpperCaseChars); err != nil {
+		return fmt.Errorf("error setting min_upper_case_chars: %w", err)
+	}
+	if err = d.Set("min_lower_case_chars", output.MinLowerCaseChars); err != nil {
+		return fmt.Errorf("error setting min_lower_case_chars: %w", err)
+	}
+	if err = d.Set("min_numeric_chars", output.MinNumericChars); err != nil {
+		return fmt.Errorf("error setting min_numeric_chars: %w", err)
+	}
+	if err = d.Set("min_special_chars", output.MinSpecialChars); err != nil {
+		return fmt.Errorf("error setting min_special_chars: %w", err)
+	}
+	if err = d.Set("max_age_days", output.MaxAgeDays); err != nil {
+		return fmt.Errorf("error setting max_age_days: %w", err)
+	}
+	if err = d.Set("max_retries", output.MaxRetries); err != nil {
+		return fmt.Errorf("error setting max_retries: %w", err)
+	}
+	if err = d.Set("lockout_time_mins", output.LockoutTimeMins); err != nil {
+		return fmt.Errorf("error setting lockout_time_mins: %w", err)
+	}
+
+	return nil
+}
+
+// UpdatePasswordPolicy implements schema.UpdateFunc.
+func UpdatePasswordPolicy(d *schema.ResourceData, meta interface{}) error {
+	manager, err := snowflake.NewPasswordPolicyManager()
+	if err != nil {
+		return fmt.Errorf("couldn't create password policy builder: %w", err)
+	}
+
+	runAlter := false
+	alterInput := &snowflake.PasswordPolicyUpdateInput{
```

Comment:
should we stick with the Snowflake naming convention? in which case this would be `PasswordPolicyAlterInput`?